src/Http/Http.Extensions/test/RequestDelegateFactoryTests.cs

```diff
@@ -5720,27 +5720,32 @@ public void Create_AddsDelegateMethodInfo_AsMetadata()
         var result = RequestDelegateFactory.Create(@delegate);
 
         // Assert
-        Assert.Contains(result.EndpointMetadata, m => m is MethodInfo);
+        // RouteHandlerEndpointDataSource adds the MethodInfo as the first item in RouteHandlerOptions.EndointMetadata
+        Assert.Empty(result.EndpointMetadata);
     }
 
     [Fact]
-    public void Create_AddsDelegateMethodInfo_AsFirstMetadata()
+    public void Create_DoesNotAddAnythingBefore_ThePassedInEndpointMetadata()
     {
         // Arrange
-        var @delegate = (AddsCustomParameterMetadata param1) => "Hello";
+        var @delegate = (AddsCustomParameterMetadataBindable param1) => { };
         var customMetadata = new CustomEndpointMetadata();
-        var options = new RequestDelegateFactoryOptions { InitialEndpointMetadata = new[] { customMetadata } };
+        var options = new RequestDelegateFactoryOptions { EndpointMetadata = new List<object> { customMetadata } };
 
         // Act
         var result = RequestDelegateFactory.Create(@delegate, options);
 
         // Assert
-        var firstMetadata = result.EndpointMetadata[0];
-        Assert.IsAssignableFrom<MethodInfo>(firstMetadata);
+        // RouteHandlerEndpointDataSource adds things like the MethodInfo, HttpMethodMetadata and attributes to RouteHandlerOptions.EndointMetadata,
+        // but we just specified our CustomEndpointMetadata in this test.
+        Assert.Collection(result.EndpointMetadata,
+            m => Assert.Same(customMetadata, m),
+            m => Assert.True(m is ParameterNameMetadata { Name: "param1" }),
+            m => Assert.True(m is CustomEndpointMetadata { Source: MetadataSource.Parameter }));
     }
 
     [Fact]
-    public void Create_AddsDelegateAttributes_AsMetadata()
+    public void Create_DoesNotAddDelegateAttributes_AsMetadata()
```

Comment:
These are interesting changes. So the caller of RDF.Create is now responsible for adding metadata, is that 6.0 behavior basically?

Reply:
Yes. We were trying to make the RDF more "helpful" in .NET 7 (I'm pretty sure this was my idea), by adding more metadata for you.

I figured few if any callers other than `EndpointRouteBuilderExtensions` bothered adding the `MethodInfo` anyway. I'm glad we didn't go through with it obviously. Because the lack of flexibility meant that `WithOpenApi()` on a group couldn't read the route handler's `MethodInfo` metadata because it hadn't beed added by RDF yet.

Reply:
Yep!